cloudfoundry/app-autoscaler-release
# Downgrade `bosh-cli` to make `bosh create-release` work in a Nix shell on macOS

# Problem
`make deploy-autoscaler` fails in the Nix shell on macOS whenever a `bosh create-release` is being executed with the following error: `'tar: unrecognized option '--no-mac-metadata'`.  

The problem here is that Nix shell provides gnutar by default, which doesn't have the `--no-mac-metadata` option.

The following change introduced the usage of `--no-mac-metadata` when `bosh create-release` runs on macOS: https://github.com/cloudfoundry/bosh-utils/commit/f79167bd43f3afc154065edc95799a464a80605f. 

# How to reproduce

1. Make sure that you have a `bosh-cli` installed on your local machine which is above `v7.3.1` by running `bosh --version`. If there is no `bosh` CLI available, install the latest one, e.g. via  `brew install bosh-cli`.
1. Open a shell and change directory into the autoscaler release repo: `cd <path-to-my-git-repos>/app-autoscaler-release`
1. Make sure that you didn't enter a Nix shell automatically (e.g. via direnv or so). In case you entered a Nix shell automatically, just run `exit` to leave it.
1. Run `bosh create-release --force --version="123"`
1. Observe that the release creation returns with the error `'tar: unrecognized option '--no-mac-metadata'`

# Solution
This PR downgrades `bosh-cli` to a version which uses a `bosh-utils` version that doesn't do `--no-mac-metadata` during a release creation. 

# How to verify if this change made `bosh create-release` work again
1. Open a shell and change directory into the autoscaler release repo: `cd <path-to-my-git-repos>/app-autoscaler-release`
1. Make sure that you entered a Nix shell.
1. Run `bosh create-release --force --version="123"`
1. Observe that the release creation returns no error like `'tar: unrecognized option '--no-mac-metadata'`

# Upstream GitHub issue
I created an issue to get this problem fixed: https://github.com/cloudfoundry/bosh-utils/issues/86. A fix for this and a new `bosh-cli` release with the fixed `bosh-utils` would allow us to get rid of the "downgrade workaround".
